File: modularisation/Clojure.hs
f x y = x + y
--f = \x y-> x + y
close x = \x->f x 5
r1 = close 6
high g = g 6
r2 = high close
r3 = high (f 5)
r4 = high f
--r5 = high (f 5 6)

File: modularisation/Clojure.java
interface Unary <R,A> { R f (A x); }
interface TwoAry <R,A,B> {R f (A x, B y); }

public class Clojure {
	public static void main (String [] args){
		final TwoAry two = new TwoAry<Integer,Integer,Integer>(){
			@Override 
			public Integer f (Integer x, Integer y) {return x + y;}
		};
		Unary close = new Unary<Integer,Integer>(){
			@Override
			public Integer f (Integer y){
				return (Integer)two.f(5,y);
			}
		};
		System.out.println(close.f(6));		
	}
}